src/data_source.js

```diff
@@ -146,9 +121,65 @@ export class NetworkTileSource extends TileSource {
         });
     }
 
-    // Sub-classes must implement this method:
-    parseSourceData (tile, source, reponse) {
-        throw new MethodNotImplemented('parseTile');
+    // Sub-classes must implement:
+
+    formatUrl (dest) {
+        throw new MethodNotImplemented('formatUrl');
+    }
+
+    parseSourceData (dest, source, reponse) {
+        throw new MethodNotImplemented('parseSourceData');
+    }
+}
+
+
+/*** Generic network tile loading - abstract class ***/
+
+export class NetworkTileSource extends NetworkSource {
+
+    constructor (source) {
+        super(source);
+
+        this.tiled = true;
+        this.url_hosts = null;
+        var host_match = this.url.match(/{s:\[([^}+]+)\]}/);
+        if (host_match != null && host_match.length > 1) {
+            this.url_hosts = host_match[1].split(',');
+            this.next_host = 0;
+        }
+    }
+
+    formatUrl(tile) {
+        var url = this.url.replace('{x}', tile.coords.x).replace('{y}', tile.coords.y).replace('{z}', tile.coords.z);
+
+        if (this.url_hosts != null) {
+            url = url.replace(/{s:\[([^}+]+)\]}/, this.url_hosts[this.next_host]);
+            this.next_host = (this.next_host + 1) % this.url_hosts.length;
+        }
+        return url;
+    }
+
+}
+
+
+/**
+ GeoJSON standalone (non-tiled) source
+*/
+
+export class GeoJSONSource extends NetworkSource {
+
+    constructor (source) {
+        super(source);
+        this.type = 'GeoJSONSource';
+    }
+
+    formatUrl (dest) {
+        return this.url;
+    }
+
+    parseSourceData (tile, source, response) {
+        source.layers = { _default: JSON.parse(response) };
```

Comment:
This is new. Is the idea to hide and keep the original source data. I could see that being useful.